spec/support/schema_matchers.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+require 'rspec/expectations'
+
+module SchemaMatchers
```

Comment:
Having to include it was a side-effect of extracting the shared code into the `valid_against_schema?` method and wrapping it all in a module. My other idea was to have a class with `valid_against_schema?` `SomeNamespace::SchemaHelper.valid_against_schema?` as a class method and then keep the `RSpec::Matchers.define :foo` global style but went with it all wrapped in the same module using the matcher DSL.